src/components/LeafletMap.svelte

```diff
@@ -0,0 +1,126 @@
+<script>
+  import deepEqual from 'deep-equal';
+  import throttle from 'lodash.throttle';
+  import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+  import * as L from 'leaflet';
+  import 'leaflet/dist/leaflet.css';
+
+  export let index;
+  export let id;
+  export let center;
+  export let zoom;
+  export let mapStyle;
+  export let numberOfMaps;
+
+  const dispatch = createEventDispatcher();
+
+  let style = {};
+  let url;
+
+  let map;
+  let layer;
+  let mapViewProps = {};
+
+  // We can set style (an object) here because mapStyle only changes when it needs to
+  $: ({ style, url } = mapStyle);
+
+  // We group map-view props here as they are useful in a few contexts
+  $: mapViewProps = { center, zoom };
+
+  const getCurrentMapView = () => {
+    return {
+      center: map.getCenter(),
+      zoom: map.getZoom() - 1,
+    };
+  };
+
+  const shouldUpdateMapView = () => {
+    return !deepEqual(getCurrentMapView(), mapViewProps);
+  };
+
+  const updateMapStyle = (map, url) => {
+    if (!map) return;
+    if (layer != null) {
+      layer.remove();
+    }
+
+    // TODO surface attribution
+    layer = L.tileLayer(url, {
+      detectRetina: true,
+    }).addTo(map);
+  };
+
+  const updateMapFromProps = (map, mapView) => {
+    if (!map || !shouldUpdateMapView(mapView)) return;
+    map.setView(mapView.center, mapView.zoom + 1, { animate: false });
+  };
+
+  onMount(async () => {
+    map = L.map(id, {
+      zoomControl: false,
+      zoomDelta: 0.25,
+      zoomSnap: 0,
```

Comment:
These three properties are making the map feel a little sluggish to me. What do you think?

Reply:
I'm not seeing sluggishness, would you mind elaborating?

Otherwise, `zoomControl: false` removes the zoom in/out buttons, `zoomSnap: 0` makes it so non-integer zooms can be used (otherwise it gets out of sync with a Mapbox GL map since the zooms don't align). So I don't think those have much to do with it but I could be wrong.

`zoomDelta` might make it feel like the zooming is slower but that would only be when you're zooming with the +/- keys.

Reply:
Thanks! I was only testing with a mouse wheel and I have a feeling you're using a trackpad--the experience is much worse with a trackpad!

Reply:
Ah, that difference didn't even occur to me! Yes, I'm using a trackpad

Reply:
I made [this example](https://jsfiddle.net/ebrelsford/b0sxzpmt/1/) with Leaflet separately and am seeing similar performance, gonna chalk it up to just how Leaflet works with trackpads if that works for you.